app/helpers/application_helper.rb

```diff
@@ -1,2 +1,116 @@
+			time = send_now ? nil : shift.start_time - min_before*60
+			title = "Upcoming Shift Reminder" if title.nil?
+			if content.nil?
+				content = "Hey! Your tent shift in K-Ville starts in #{min_before} minutes!"
+			end
+			puts "Scheduling shift notification with time: #{time} title: #{title} content: #{content} netIDs: #{netids.join(', ')}"
+			onesignal_id = create_notification(netids, recipient_type='netids',
+																				 title, content, time, test)
+			# Store notification record in db for each user
+			userids = shift.users.collect(&:id)
+			userids.each do |id|
+				@user = User.find(id)
+				@user.notifications.create(notification_id: onesignal_id, start_time: time, title: title, content: content)
+			end
+			#puts @user.notifications.pluck(:start_time, :title, :content)
+			onesignal_id
+		end
+
+		# @note Helper method. Can delete if necessary but allows for additional
+		# logic for post notifications.
+		# Currently, post notifications send immediately to all users, meant to
+		# be used for line monitor announcements.
+		def post_notification(title, content, test: false)
+			# Send to all members, immediately
+			puts "Scheduling post announcement notification to all users with title: #{title} content: #{content}"
+			onesignal_id = create_notification(['All'], recipient_type='all',
+																				 title, content, time=nil, test)
+			#puts @user.notifications.pluck(:start_time, :title, :content)
+			onesignal_id
+		end
+
+    # @note Function to create a scheduled notification for a list of users
+    # by netid, which is mapped to a OneSignal device by an external ID.
+		# @param [Array] recipients
+    # @param [String] recipient_type, one of (netids, all, players)
+		# @param [String] title of notification
+		# @param [String] content of notification
+    # @param [Boolean] test parameters without hitting OneSignal API.
+		# @return [String] OneSignal ID of scheduled notification
+		def create_notification(recipients, recipient_type, title, content, time,
+														test)
+			if recipient_type == 'netids'
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									'include_external_user_ids' => recipients}
+			elsif recipient_type == 'all'
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									 'included_segments' => ['All']}
+			elsif recipient_type == 'players'
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									'include_player_ids' => recipients}
+			else # assume sending to test users
+				params = {'app_id' => 'b290fd9a-eedf-44b0-8bfd-6a37646957b6',
+									'headings' => {'en' => title}, 'contents' => {'en' => content},
+									'included_segments' => ['Test Users']}
```

Comment:
Fix tabbing and spacing on these lines. Too much content on one line, hurt readability.

Reply:
Fixed tabs I think

Reply:
Actually my IDE is being weird and just visually showing me changes in my editor but it's not affecting the actual file. Will get this fixed.